.github/workflows/docker-bases.yml

```diff
@@ -0,0 +1,90 @@
+name: Build base compilers docker images
+
+on:
+  push:
+    paths:
+      - './docker/Dockerfile.nvidia'
+      - './docker/Dockerfile.cpu'
+      - './docker/Dockerfile.amd'
```

Comment:
If I'm understanding this correctly, then if any one of three docker files is modified then all three are rebuilt? If so, then perhaps these should be 3 different actions?

Reply:
Yes it's a bit overkill but the hope is that those bases will very rarely be touched so it's "acceptable". Three workflow would be a bit of a pain but can look into a nicer dispatch